README.md

```diff
@@ -278,13 +278,24 @@ the following parts are still incomplete:
   this kind of uri does not give any information about the location of a
   corresponding file on disk.
 
-- Type arguments of generic types are only supported in some simple cases,
-  because we do not have the primitives required for a general
-  implementation. E.g., when it is known that a given list of actual type
-  arguments is empty then the empty list is returned. However, when a
-  parameterized type has a non-trivial list of actual type arguments then
-  returning the actual type arguments would require runtime support that does
-  not currently exist.
+- Type arguments of generic types are only supported in some statically
+  resolved cases, because we do not have the primitives required for a
+  general implementation. E.g., when it is known statically that a given
+  list of actual type arguments is empty then the empty list is
+  returned. Moreover, if support for reflected types is included and a type
+  annotation contains only types which are fully resolved statically, it is
+  possible to get these type arguments. So, for instance, with a field of
+  type `Map<int, String>` and suitable capabilities, we can invoke
```

Comment:
I would write the examples in a long form to show full examples of what is supported and what is not. I think that would ease the comprehension. Maybe along the lines of:
````
```dart
class Reflector extends Reflectable {
  const Reflector() : super(declarationsCapability, typeCapability, typeRelationsCapability, reflectedTypeCapability);
}

@Reflector()
class A<T> {
  Map<int, String> map;
  List<X> list;
}

main() {
  initializeReflectable();
  final declarations = reflector.reflectType(A).declarations;
  declarations['map'].reflectedTypeArguments[0]; // Will return int.
  declarations['list'].reflectedTypeArguments[0]; // Will throw at runtime.
}
```
````

Reply:
Done!